.github/workflows/test_cpp.yml

```diff
@@ -45,7 +45,7 @@ jobs:
             cache_key: Bazel7
             image: "us-docker.pkg.dev/protobuf-build/containers/common/linux/bazel:7.1.2-cf84e92285ca133b9c8104ad7b14d70e953cbb8e"
             targets: "//src/... //third_party/utf8_range/..."
-          - config: { name: "Bazel7 with Bzlmod", flags: --enable_bzlmod --enable_workspace }
+          - config: { name: "Bazel7 with Bzlmod", flags: --enable_bzlmod --enable_workspace --cxxopt=-std=c++17 --host_cxxopt=-std=c++17 --cxxopt="-Wno-self-assign-overloaded" }
```

Comment:
This should ideally stay on c++14 in 29.x, if silencing the warnings without bumping to Bazel 17 is sufficient here. We're not raising our minimum C++ version to 17 until 30.x

Reply:
I can't silence the warning, because it's enabled on abseil:
https://github.com/abseil/abseil-cpp/blob/master/absl/strings/BUILD.bazel#L91

Reply:
Done. Disable using --per_file_copt